stackstate_checks_base/stackstate_checks/base/checks/base.py

```diff
@@ -667,28 +669,48 @@ def _get_requests_proxy(self):
         return proxies if proxies else no_proxy_settings
 
     # TODO collect all errors instead of the first one
-    def _fix_encoding(self, value, context=None):
-        if isinstance(value, text_type):
+    def _sanitize(self, field, context=None):
+        """
+        Fixes encoding and strips empty elements.
+        :param field: Field can be of the following types: str, dict, list, set
+        :param context: Context for error message.
+        :return:
+        """
+        if isinstance(field, text_type):
             try:
-                fixed_value = to_string(value)
+                fixed_value = to_string(field)
             except UnicodeError as e:
-                self.log.warning("Error while encoding unicode to string: '{0}', at {1}".format(value, context))
+                self.log.warning("Error while encoding unicode to string: '{0}', at {1}".format(field, context))
                 raise e
             return fixed_value
-        elif isinstance(value, dict):
-            for key, field in list(iteritems(value)):
-                value[key] = self._fix_encoding(field, "key '{0}' of dict".format(key))
-        elif isinstance(value, list):
-            for i, element in enumerate(value):
-                value[i] = self._fix_encoding(element, "index '{0}' of list".format(i))
-        elif isinstance(value, set):
+        elif isinstance(field, dict):
+            field = {k: v for k, v in iteritems(field) if self._is_field_empty(v)}
+            for key, value in list(iteritems(field)):
+                field[key] = self._sanitize(value, "key '{0}' of dict".format(key))
+        elif isinstance(field, list):
+            field = [element for element in field if self._is_field_empty(element)]
+            for i, element in enumerate(field):
+                field[i] = self._sanitize(element, "index '{0}' of list".format(i))
+        elif isinstance(field, set):
             # we convert a set to a list so we can update it in place
             # and then at the end we turn the list back to a set
-            encoding_list = list(value)
+            encoding_list = [element for element in list(field) if self._is_field_empty(element)]
             for i, element in enumerate(encoding_list):
-                encoding_list[i] = self._fix_encoding(element, "element of set")
-            value = set(encoding_list)
-        return value
+                encoding_list[i] = self._sanitize(element, "element of set")
+            field = set(encoding_list)
+        return field
+
+    def _is_field_empty(self, field):
```

Comment:
maybe I'm too tired but shouldn't this function be called the **opposite**, thus `_is_not_empty` ?

because only if not empty we will retain it during sanitization